.gitignore

```diff
@@ -1,36 +1,57 @@
+# Quick format and style primer:
+#
+#  * If a pattern is meant only for a specific location, it should have a
+#    leading slash, like `/staticfiles.json`.
+#    * In principle any non-trailing slash (like `zproject/dev-secrets.conf`)
+#      will do, but this makes a confusing pattern.  Adding a leading slash
+#      is clearer.
+#
+#  * Patterns like `.vscode/` without slashes, or with only a trailing slash,
+#    match in any subdirectory.
+#
+#  * Subdirectories with several internal things to ignore get their own
+#    `.gitignore` files.
+#
+# See `git help ignore` for details on the format.
+
+## Config files for the dev environment
+/zproject/dev-secrets.conf
+/tools/conf.ini
+/tools/custom_provision
+
+## Byproducts of setting up and using the dev environment
 *.pyc
-*~
+
+/queue_error
+/.vagrant
+/var
+
+# Static build
+*.mo
+npm-debug.log
+/node_modules
 /prod-static
+/staticfiles.json
+
+# Test / analysis tools
+.mypy_cache/
+.coverage
+/coverage/
```

Comment:
Hmm, that was a little careless of me -- I translated `coverage/` to `/coverage/` in this rearrangement commit.

Searching the tree shows references to a `docs/_build/coverage/`, a `var/coverage/`, and a dev URL subtree `/coverage/` which maps to the latter. Both of those should be covered by other patterns, so I'll drop this one.